src/persist-client/src/internal/maintenance.rs

```diff
@@ -30,6 +32,53 @@ pub struct LeaseExpiration {
     pub(crate) writers: Vec<WriterId>,
 }
 
+impl LeaseExpiration {
+    pub(crate) async fn perform<K, V, T, D>(
+        self,
+        machine: &Machine<K, V, T, D>,
+    ) -> RoutineMaintenance
+    where
+        K: Debug + Codec,
+        V: Debug + Codec,
+        T: Timestamp + Lattice + Codec64,
+        D: Semigroup + Codec64,
+    {
+        let expire_readers = self.readers.into_iter().map(|expired| {
+            let mut machine = machine.clone();
+            mz_ore::task::spawn(|| "persist::automatic_read_expiration", async move {
+                info!(
+                    "Force expiring reader ({}) of shard ({}) due to inactivity",
+                    expired,
+                    machine.shard_id()
+                );
+                // XXX This ought to return maintenance too... along with most of the other
+                // explicit ::default() calls in this file.
```

Comment:
yeah there's quite a few 😓

Reply:
oh I think I misread. comment was made in reference to the # of machine calls where we swallow maintenance right now

Reply:
Yes, second comment is right! (All the XXX are "things that I'd need to fix before a real PR", so any commentary is definitely aimed at the WIP and not existing stuff.)